bundles/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/values/NumberValue.java

```diff
@@ -145,8 +152,9 @@ public StateDescriptionFragmentBuilder createStateDescription(boolean readOnly)
             builder = builder.withMinimum(min);
         }
         builder = builder.withStep(step);
-        if (this.unit.length() > 0) {
-            builder = builder.withPattern("%s " + this.unit.replace("%", "%%"));
+        String unitSymbol = this.unit.getSymbol();
```

Comment:
Could it be enough to just do
```
         return builder.withStep(step).withPattern("%s %unit%");
```
now, that we switched to QuantityType?

Reply:
I have to say I don't know much about StateDescriptionFragmentBuilder and I've struggled to track down any docs on it. It seems to me that most bindings don't implement it at all. I think we have two options, what you suggest (which improves the PR, and I think maintains compatibility with existing configurations), or to remove all calls to `withPattern` (don't call the super method either), which seems to leave rendering up to the default behaviour for the state. In that scenario everything appears as I expect (with units) in the MainUI, but Sitemaps expect a state presentation format in order to display anything, so existing users might be confused by values disappearing. The only benefit I can think of is some localisation where (e.g.) units are presented before the number, but can't find any evidence that exists, so probably easiest to go with your proposal?